src/transport/assembly_dg.hh

```diff
@@ -569,69 +573,81 @@ public:
 
-                double transport_flux = arma::dot(eq_fields_->advection_coef[sbi](p_high), fe_values_side_.normal_vector(k));
-
-                comm_flux[0][0] =  (sigma-min(0.,transport_flux))*fv_sb_[0]->JxW(k);
-                comm_flux[0][1] = -(sigma-min(0.,transport_flux))*fv_sb_[0]->JxW(k);
-                comm_flux[1][0] = -(sigma+max(0.,transport_flux))*fv_sb_[0]->JxW(k);
-                comm_flux[1][1] =  (sigma+max(0.,transport_flux))*fv_sb_[0]->JxW(k);
-
-                for (int n=0; n<2; n++)
-                {
-                    if (!own_element_id[n]) continue;
-
-                    for (unsigned int i=0; i<n_dofs[n]; i++)
-                        for (int m=0; m<2; m++)
-                            for (unsigned int j=0; j<n_dofs[m]; j++)
-                                local_matrix_[(i+n*n_dofs[0])*(n_dofs[0]+n_dofs[1]) + m*n_dofs[0] + j] +=
-                                        comm_flux[m][n]*fv_sb_[m]->shape_value(j,k)*fv_sb_[n]->shape_value(i,k) + LocalSystem::almost_zero;
+                double transport_flux = arma::dot(eq_fields_->advection_coef[sbi](p_high), normal_(p_high));
+
+//                comm_flux[0][0] =  (sigma-min(0.,transport_flux))*JxW_(p_high);
+//                comm_flux[0][1] = -(sigma-min(0.,transport_flux))*JxW_(p_high);
+//                comm_flux[1][0] = -(sigma+max(0.,transport_flux))*JxW_(p_high);
+//                comm_flux[1][1] =  (sigma+max(0.,transport_flux))*JxW_(p_high);
+//
+//                for (int n=0; n<2; n++)
+//                {
+//                    if (!own_element_id[n]) continue;
+//
+//                    for (unsigned int i=0; i<n_dofs[n]; i++)
+//                        for (int m=0; m<2; m++)
+//                            for (unsigned int j=0; j<n_dofs[m]; j++)
+//                                local_matrix_[(i+n*n_dofs[0])*(n_dofs[0]+n_dofs[1]) + m*n_dofs[0] + j] +=
+//                                        comm_flux[m][n]*fv_sb_[m]->shape_value(j,k)*fv_sb_[n]->shape_value(i,k) + LocalSystem::almost_zero;
+//                }
+//                k++;
+                for( auto conc_shape_i : conc_join_shape_) {
+                    uint is_high_i = conc_shape_i.is_high_dim();
+                    if (!own_element_id[is_high_i]) continue;
+                    uint i_mat_idx = conc_shape_i.join_idx(); // i + is_high * n_dofs_low
+                    double diff_shape_i = conc_shape_i(p_high) - conc_shape_i(p_low_conv);
+                    for( auto conc_shape_j : conc_join_shape_) {
+                        uint j_mat_idx = conc_shape_j.join_idx();
+                        local_matrix_[i_mat_idx * (n_dofs[0]+n_dofs[1]) + j_mat_idx] += (
+                                sigma * diff_shape_i * (conc_shape_j(p_high) - conc_shape_j(p_low_conv))
+                                + diff_shape_i * ( max(0.,transport_flux) * conc_shape_j(p_high) - min(0.,transport_flux) * conc_shape_j(p_low_conv))
```

Comment:
Not sure if fv_sb_[0] corresponds to p_high and fv_sb_[1] to p_low.
If so, then there should be rather diff_shape_i * (-max * shape_j(p_low) - min * shape_j(p_high)

Anyway, seems ther should be the same sign for min and max.